zerver/lib/users.py

```diff
@@ -578,16 +601,21 @@ def get_subscribers_of_target_user_subscriptions(
         target_user_subscriptions_dict[user_profile_id] = recipient_ids
         target_users_subbed_recipient_ids |= recipient_ids
 
-    subs_in_target_user_subscriptions = (
-        Subscription.objects.filter(
-            recipient_id__in=list(target_users_subbed_recipient_ids),
-            active=True,
-            recipient__type__in=[Recipient.STREAM, Recipient.HUDDLE],
-        )
-        .order_by("recipient_id")
-        .values("user_profile_id", "recipient_id")
+    subs_in_target_user_subscriptions_query = Subscription.objects.filter(
+        recipient_id__in=list(target_users_subbed_recipient_ids),
+        active=True,
+        recipient__type__in=[Recipient.STREAM, Recipient.HUDDLE],
```

Comment:
Adding code to include only active users here makes this function somewhat complicated. One possible workaround would be to filter out inactive users in callers of these functions wherever required in a simpler way like just taking intersection with `active_user_ids(realm.id)` and in some cases this would not be required since we already filter out inactive users.

But having an option to exclude deactivated users seems better logically to me even though it makes it somewhat hard to read.